src/pages/EditRequestPage.js

```diff
@@ -115,26 +111,18 @@ function EditRequestPage({betas, report, route, parentReport, policyCategories,
 
     // Decides whether to allow or disallow editing a money request
     useEffect(() => {
-        const isNonEditableFieldWhenSettled = _.includes(
-            [CONST.EDIT_REQUEST_FIELD.AMOUNT, CONST.EDIT_REQUEST_FIELD.DATE, CONST.EDIT_REQUEST_FIELD.RECEIPT, CONST.EDIT_REQUEST_FIELD.DISTANCE],
-            fieldToEdit,
-        );
+        const canEdit = ReportUtils.canEditFieldOfMoneyRequest(parentReportAction, parentReport.reportID, fieldToEdit);
 
-        /**
-         * Do not dismiss the modal, when a current user can edit a money request.
-         * To satisfy the condition below:
-         * 1. "canEdit" must be "true". It checks common rules like if the user is a requestor or admin, etc.
-         * 2. When the current edit field is the "amount" or "date" (created date), the money request shouldn't be settled yet.
-         * */
-        if (canEdit && (!isNonEditableFieldWhenSettled || !isSettled)) {
+        // Do not dismiss the modal, when a current user can edit a money request.
+        if (canEdit) {
```

Comment:
The `canEdit` variable isn't necessary since it's only referenced once. Please change this in the other file as well.

Reply:
@tgolen, do you mean something like that - c2bfd64?